base/pkg/cache.jl

```diff
@@ -28,27 +28,35 @@ function mkcachedir()
     mkdir(cache)
 end
 
-
 function prefetch(pkg::AbstractString, url::AbstractString, sha1s::Vector)
     isdir(".cache") || mkcachedir()
+    #TODO: force switch to https
+    #url = LibGit2.normalize_url(url)
     cache = path(pkg)
-    if !isdir(cache)
+    repo = if isdir(cache)
+        LibGit2.GitRepo(cache) # open repo, free it at the end
+    else
         info("Cloning cache of $pkg from $url")
-        try Git.run(`clone -q --mirror $url $cache`)
-        catch
-            rm(cache, recursive=true)
-            rethrow()
+        try
+            # clone repo, free it at the end
+            LibGit2.clone(url, cache, bare = true, remote_cb = LibGit2.mirror_cb)
+        catch err
+            isdir(cache) && rm(cache, recursive=true)
+            throw(PkgError("Cannot clone $pkg from $url. $(err.msg)"))
         end
     end
-    Git.set_remote_url(url, dir=cache)
-    in_cache = Git.iscommit(sha1s, dir=cache)
-    if !all(in_cache)
-        info("Updating cache of $pkg...")
-        Git.success(`remote update`, dir=cache) ||
-        error("couldn't update $cache using `git remote update`")
```

Comment:
Potentially worth noting - IIUC - that this is an API change: this no longer errors if there is a problem with fetch (LibGit2.fetch just warns). New behaviour is better IMO.